File: backend/src/main/java/tom/projects/monkeylog/dto/workout/ExerciseCategoryResponse.java
package tom.projects.monkeylog.dto.workout;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.Data;
import tom.projects.monkeylog.model.exercise.ExerciseCategory;

import java.util.List;

@Data
@Builder
public class ExerciseCategoryResponse {
    @NotEmpty
    private ExerciseCategory id;
    @NotEmpty
    private String name;
    @NotNull
    private List<ExerciseTypeResponse> exerciseTypes;
}

File: backend/src/main/kotlin/tom/com/monkeylog/dto/workout/ExerciseCategoryResponse.kt
package tom.com.monkeylog.dto.workout

import tom.com.monkeylog.dto.statistics.StatisticsType
import tom.com.monkeylog.model.exercise.ExerciseCategory

data class ExerciseCategoryResponse(
    val statistics: List<StatisticsType>,
    val validFields: ExerciseCategory.ValidFields
)
